java/ChartboostMediationDemoJava/src/main/java/com/example/chartboost/mediation/sdk/demo/java/AdController.java

```diff
@@ -0,0 +1,296 @@
+package com.example.chartboost.mediation.sdk.demo.java;
+
+import android.app.Activity;
+import android.content.Context;
+import android.view.View;
+import android.widget.Button;
+import android.widget.FrameLayout;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.chartboost.chartboostmediationsdk.ChartboostMediationSdk;
+import com.chartboost.chartboostmediationsdk.ad.AdLoadResult;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationAdShowResult;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationBannerAdLoadListener;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationBannerAdLoadRequest;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationBannerAdLoadResult;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationBannerAdView;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationBannerAdViewListener;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAd;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdListener;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdLoadListener;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdLoadRequest;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdLoadResult;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdQueue;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdQueueListener;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdQueueManager;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdShowListener;
+import com.chartboost.chartboostmediationsdk.domain.ChartboostMediationAdException;
+import com.chartboost.chartboostmediationsdk.domain.Keywords;
+import com.chartboost.core.ChartboostCore;
+import com.chartboost.core.initialization.ModuleInitializationResult;
+import com.chartboost.core.initialization.ModuleObserver;
+import com.chartboost.core.initialization.SdkConfiguration;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class AdController {
+
+    private static final int UNCONFIGURED_AD_QUEUE_CAPACITY = 5;
+    private final Map<String, ChartboostMediationFullscreenAd> fullscreenAds = new ConcurrentHashMap();
```

Comment:
this is super overkill if you only ever use two placements, but it's fine to have it to showcase that you can do this (assuming other people will be using multiple placements).
however, you don't showcase how in an ad queue setting, you can just chain shows.
You can get into a situation where an interstitial loads and an extra one caches. on show finished, the next ad is already ready to show. you don't need to wait for the next one to cache.